src/App.tsx

```diff
@@ -1,8 +1,135 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { DownloadTask } from './modules/DownloadTask';
+import { RenderTaskList } from './modules/RenderTaskList';
+import { Prepared, User, Task } from './modules/interfaces';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const tasksLink = 'https://mate.academy/students-api/todos';
+const usersLink = 'https://mate.academy/students-api/users';
+
+const App = () => {
+  const [prepared, setPrepared] = useState<Prepared[]>([]);
+  const [isDataDownloaded, setDataDownloaded] = useState(false);
+  const [buttonText, setButtonText] = useState('Download tasks');
+
+  async function fetchData<T>(url: string): Promise<T> {
+    const data = await fetch(url);
+    const response = await data.json();
+
+    return response.data;
+  }
+
+  async function getData() {
+    const tasks = await fetchData<Task[]>(tasksLink);
+    const users = await fetchData<User[]>(usersLink);
+
+    setPrepared(
+      tasks.map((task) => {
+        return {
+          ...task,
+          user: users.find(person => person.id === task.userId),
+        } as Prepared;
+      }),
+    );
+
+    setDataDownloaded(true);
+  }
+
+  const sorting = (sortType: string) => {
+    switch (sortType) {
+      case 'title':
+        return setPrepared(
+          [...(prepared as Prepared[])
```

Comment:
этого не должно тут быть
```suggestion
          [...prepared)
```

Reply:
✓